flexget/plugins/filter/series.py

```diff
@@ -1961,14 +1958,11 @@ def process_propers(self, config, episode, entries):
         """
 
         pass_filter = []
-        best_propers = []
-        # Since eps is sorted by quality then proper_count we always see the highest proper for a quality first.
-        (last_qual, best_proper) = (None, 0)
+        # First find best available proper for each quality without modifying incoming entry order
+        sorted_entries = sorted(entries, key=lambda e: (e['quality'], e['proper_count']), reverse=True)
+        best_propers = {q: next(e) for q, e in itertools.groupby(sorted_entries, key=lambda e: e['quality'])}
```

Comment:
god help who has to debug this one liner 😆

Reply:
That's why I split it into a 2 liner. 😉